sybilMTP.go

```diff
@@ -0,0 +1,261 @@
+	StateCommitmentClaimRevRoot    *merkletree.Hash   `json:"stateCommitmentClaimRevRoot"`
+	StateCommitmentClaimRootsRoot  *merkletree.Hash   `json:"stateCommitmentClaimRootsRoot"`
+	StateCommitmentClaimIdenState  *merkletree.Hash   `json:"stateCommitmentClaimIdenState"`
+
+	GistRoot     *merkletree.Hash   `json:"gistRoot"`
+	GistMtp      []*merkletree.Hash `json:"gistMtp"`
+	GistMtpAuxHi *merkletree.Hash   `json:"gistMtpAuxHi"`
+	GistMtpAuxHv *merkletree.Hash   `json:"gistMtpAuxHv"`
+	GistMtpNoAux string             `json:"gistMtpNoAux"`
+
+	CRS *big.Int `json:"crs"`
+
+	UserGenesisID            string `json:"userGenesisID"`
+	ProfileNonce             string `json:"profileNonce"`
+	ClaimSubjectProfileNonce string `json:"claimSubjectProfileNonce"`
+
+	RequestID *big.Int `json:"requestID"`
+	IssuerID  string   `json:"issuerID"`
+	Timestamp int64    `json:"timestamp"`
+}
+
+func (s SybilMTPInputs) Validate() error {
+
+	if s.ID == nil {
+		return errors.New(ErrorEmptyID)
+	}
+
+	if s.GISTProof.Proof == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	if s.IssuerClaim.Claim == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	if s.StateCommitmentClaim.Claim == nil {
+		return errors.New(ErrorEmptyGlobalProof)
+	}
+
+	return nil
+}
+
+func (s SybilMTPInputs) InputsMarshal() ([]byte, error) {
+	if err := s.Validate(); err != nil {
+		return nil, err
+	}
+
+	mtpInputs := sybilMTPCircuitInputs{
+		IssuerClaim:           s.IssuerClaim.Claim,
+		IssuerClaimMtp:        CircomSiblings(s.IssuerClaim.IncProof.Proof, s.GetMTLevel()),
+		IssuerClaimClaimsRoot: s.IssuerClaim.IncProof.TreeState.ClaimsRoot,
+		IssuerClaimRevRoot:    s.IssuerClaim.IncProof.TreeState.RevocationRoot,
+		IssuerClaimRootsRoot:  s.IssuerClaim.IncProof.TreeState.RootOfRoots,
+		IssuerClaimIdenState:  s.IssuerClaim.IncProof.TreeState.State,
+
+		IssuerClaimNonRevMtp: CircomSiblings(s.IssuerClaim.NonRevProof.Proof, s.GetMTLevel()),
+
+		IssuerClaimNonRevClaimsRoot: s.IssuerClaim.NonRevProof.TreeState.ClaimsRoot,
+		IssuerClaimNonRevRevRoot:    s.IssuerClaim.NonRevProof.TreeState.RevocationRoot,
+		IssuerClaimNonRevRootsRoot:  s.IssuerClaim.NonRevProof.TreeState.RootOfRoots,
+		IssuerClaimNonRevState:      s.IssuerClaim.NonRevProof.TreeState.State,
+
+		ClaimSchema: s.IssuerClaim.Claim.GetSchemaHash().BigInt().String(),
+
+		StateCommitmentClaim:           s.StateCommitmentClaim.Claim,
+		StateCommitmentClaimMtp:        CircomSiblings(s.StateCommitmentClaim.IncProof.Proof, s.GetMTLevel()),
+		StateCommitmentClaimClaimsRoot: s.StateCommitmentClaim.IncProof.TreeState.ClaimsRoot,
+		StateCommitmentClaimRevRoot:    s.StateCommitmentClaim.IncProof.TreeState.RevocationRoot,
+		StateCommitmentClaimRootsRoot:  s.StateCommitmentClaim.IncProof.TreeState.RootOfRoots,
+		StateCommitmentClaimIdenState:  s.StateCommitmentClaim.IncProof.TreeState.State,
+
+		GistRoot: s.GISTProof.Root,
+		GistMtp:  CircomSiblings(s.GISTProof.Proof, s.GetMTLevel()),
```

Comment:
For Gist levels, use this one `s.GetMTLevelOnChain`
GISTMtp: merkletree.CircomSiblingsFromSiblings(a.GISTProof.Proof.AllSiblings(),
			a.GetMTLevelOnChain()-1),
	}